src/messaging/ReliableMessageMgr.cpp

```diff
@@ -353,6 +354,17 @@ CHIP_ERROR ReliableMessageMgr::SendFromRetransTable(RetransTableEntry * entry)
 
     if (err == CHIP_NO_ERROR)
     {
+        const ExchangeManager * exchangeMgr = entry->ec->GetExchangeMgr();
+        // TODO: investigate why in ReliableMessageMgr::CheckResendApplicationMessageWithPeerExchange unit test released exchange
+        // context with mExchangeMgr==nullptr is used.
```

Comment:
@pan-apple Can you answer the TODO ?

Reply:
The problem I met was failing `CheckResendApplicationMessageWithPeerExchange` test in the CI. After further investigation it turned out to be segmentation fault, because of accessing `nullptr` on calling `entry->ec->GetExchangeMgr();`. I checked that test uses in this place in the code the `ExchangeContext` after calling its destructor, what doesn't happen to me in the regular communication or other tests. Of course we can just check pointer before accessing it, but it doesn't feel right to use destroyed object.